chain/network/src/peer/peer_actor.rs

```diff
@@ -1732,12 +1741,18 @@ impl actix::Handler<stream::Frame> for PeerActor {
 
         tracing::trace!(target: "network", "Received message: {}", peer_msg);
 
+        let now = self.clock.now();
         {
             let labels = [peer_msg.msg_variant()];
             metrics::PEER_MESSAGE_RECEIVED_BY_TYPE_TOTAL.with_label_values(&labels).inc();
             metrics::PEER_MESSAGE_RECEIVED_BY_TYPE_BYTES
                 .with_label_values(&labels)
                 .inc_by(msg.len() as u64);
+            if !self.received_messages_rate_limits.is_allowed(&peer_msg, now) {
+                metrics::PEER_MESSAGE_RATE_LIMITED_BY_TYPE_TOTAL.with_label_values(&labels).inc();
+                tracing::debug!(target: "network", "Peer {} is being rate limited for message {}", self.peer_info, peer_msg.msg_variant());
```

Comment:
Food for thought, what purpose do you envision this log message serving?

I guess one thing it helps us to understand is *which* peer is sending the spam, something which the aggregate metrics (which are only labelled by variant) do not capture. Another way to check which peer is spamming us could be to look at the receive rate (in bytes) per connection, something which is reported via debug api and visible on debug pages. However, that would not be fine-grained by message variant.

Just trying to think through how this might be used and whether there might be a more efficient way to represent the same information.